rust/sys/src/sys.rs

```diff
@@ -0,0 +1,54 @@
+#[repr(u32)]
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub enum AppProtoEnum {
+    ALPROTO_UNKNOWN = 0,
+    ALPROTO_FAILED = 1,
+    ALPROTO_HTTP1 = 2,
+    ALPROTO_FTP = 3,
```

Comment:
Could also do:
```
pub const AppProtoEnum_ALPROTO_UNKNOWN: AppProtoEnum = 0;
pub const AppProtoEnum_ALPROTO_FAILED: AppProtoEnum = 1;
pub const AppProtoEnum_ALPROTO_HTTP1: AppProtoEnum = 2;
pub const AppProtoEnum_ALPROTO_FTP: AppProtoEnum = 3;
```

Reply:
Sure? That PR still uses the `AppProto` datatype in SNMP. Make sense to bindgen that from C to Rust for use.

Reply:
Indeed I see `AppProtoEnum as AppProto` https://github.com/OISF/suricata/pull/12466#discussion_r1928380883

Some comments around it may help like 
- AppProtoEnum is for statically defined protocols
- AppProto is for all protocols : static and dynamic

Reply:
This file is generated, so I guess that doc would go where these are defined in C.